django/http/request.py

```diff
@@ -257,8 +261,19 @@ def body(self):
         if not hasattr(self, '_body'):
             if self._read_started:
                 raise RawPostDataException("You cannot access body after reading from request's data stream")
+
+            # Limit the maximum request data size that will be handled in-memory.
             try:
-                self._body = self.read()
+                content_length = int(self.META.get('HTTP_CONTENT_LENGTH', self.META.get('CONTENT_LENGTH', 0)))
+            except (ValueError, TypeError):
+                content_length = 0
+
+            if (settings.DATA_UPLOAD_MAX_MEMORY_SIZE is not None
+                    and content_length > settings.DATA_UPLOAD_MAX_MEMORY_SIZE):
+                raise SuspiciousOperation('Request data too large')
+
+            try:
+                self._body = self.read(size=content_length)
```

Comment:
Any reason for limiting to content_length here? It was not there before and does not improve the security of it in any way…